Sources/SwiftDocC/Infrastructure/Link Resolution/PathHierarchy.swift

```diff
@@ -102,7 +102,12 @@ struct PathHierarchy {
             var nodes: [String: Node] = [:]
             nodes.reserveCapacity(graph.symbols.count)
             for (id, symbol) in graph.symbols {
-                if let existingNode = allNodes[id]?.first(where: { $0.symbol!.identifier == symbol.identifier }) {
+                if let existingNode = allNodes[id]?.first(where: {
+                    // If both identifiers are in the same language, they are the same symbol
+                    $0.symbol!.identifier.interfaceLanguage == symbol.identifier.interfaceLanguage
```

Comment:
The path hierarch is initialized with only symbol data so throughout the initialization there are never any articles, tutorials, or on-page landmarks to worry about. Other types of content are added to the path hierarchy after its initialization. This resembles how content is registered with the documentation context in phases.

There is however one situation where a "sparse" node is created to fill in missing parents for the convert service but those nodes are never added to `allNodes` so it holds true that every node _in allNodes_ always has a symbol.

A "sparse" node is different from other nodes because it doesn't have a node identifier. It is considered "unfindable", so it's also not in the `lookup`, but it needs to exist in the hierarchy to traverse the tree up or down.